### PR TITLE
git-rebase: add --onto option

### PR DESCRIPTION
**Pull request by liuderchi**
`git rebase --onto <NEW_BASE> <OLD_BASE>` is a common usage if you want to keep history tidy.

### CODE REVIEW

File: pages/common/git-rebase.md [hunk 1]
```diff
@@ -17,3 +17,7 @@
 - Abort a rebase in-progress:
 
 `git rebase --abort`
+
+- Rebase your local branch by specifying new base commit and old base commit:
+
+`git rebase --onto {{NEW-BASE-COMMIT}} {{OLD-BASE-COMMIT}}`
```
Comment (waldyrious) [inline pages/common/git-rebase.md]: Please use lowercase and underscore in the tokens.

Comment (liuderchi) [inline pages/common/git-rebase.md]: ok.
And found that `git-push.md` and `git-merge.md` not follow this style.

Comment (agnivade) [inline pages/common/git-rebase.md]: Yes, we came up with a finalized style guide a bit later. You can fix those too in another PR if you would like to :smiley:

Comment (liuderchi) [inline pages/common/git-rebase.md]: I'd love to :smiley: 
Maybe it's a whole-scan through `*.md` files.
And let me confirm that the token syntax guide is [here](https://github.com/tldr-pages/tldr/blob/master/CONTRIBUTING.md#token-syntax), right?

Comment (waldyrious) [inline pages/common/git-rebase.md]: Exactly right @liuderchi. And a full-repo scan to fix tokens would be most welcome!

**Comment by waldyrious** [comment]
One question, @liuderchi: would `git rebase --onto {{new_base_commit}}` (without specifying the `{{old_base_commit}}`) be the same as `git rebase {{new_base_commit}}`?

**Comment by liuderchi** [comment]
> would git rebase --onto {{new_base_commit}} (without specifying the {{old_base_commit}}) be the same as git rebase {{new_base_commit}}?

@waldyrious  the ans is _no_ is most cases. (unless you config git with something)

TLDR：
- `--onto` option is used when you want _special base commit of current branch_, 
- otherwise use `git rebase [-i] master` is fine :+1: 


According `man git rebase` paragraph 2nd:
> If \<upstream\> is not specified, the upstream configured in
       branch.\<name\>.remote and branch.\<name\>.merge options will be used (see
       git-config(1) for details)

And in default it's not configured so this command fails and did nothing.

What's more, if you append _current branch_ (e.g. _myfeature_) as argument to the command
the result of the git history will be _only the current branch points to master_
and _commits in myfeature_ is lost.

**Comment by waldyrious** [comment]
I may be reading you incorrectly, but "no in most cases" seems to contradict with the TLDR that you present afterwards. In any case, the change itself looks good to me. I'll revisit once I learn more about `rebase --onto`.

**Comment by waldyrious** [comment]
LGTM

**Comment by liuderchi** [comment]
@waldyrious sorry for confusing words I used.

* "no in most cases" means: 
    - without `{{old_base_commit}}` git would try to look up your git config, 
    - but in _default_ the target config entry is NOT set. In other words I just can't promise the ans is no in any cases.

* the previous TLDR section is for general scenarios, which is more straight forward.

**Comment by agnivade** [comment]
:+1:
